adapters/Maio/MaioAdapter/RTBAdapter/GADRTBMaioInterstitialAd.m

```diff
@@ -0,0 +1,73 @@
+//
+//  GADRTBMaioInterstitialAd.m
+//  Adapter
+//
+//  Created by i-mobile on 2020/11/18.
+//  Copyright © 2020 Google. All rights reserved.
+//
+
+#import "GADRTBMaioInterstitialAd.h"
+#import "GADMMaioConstants.h"
+
+#import <MaioOB/MaioOB-Swift.h>
+
+#define MaioInterstitial MaioRewarded
+#define MaioInterstitialLoadCallback MaioRewardedLoadCallback
+#define MaioInterstitialShowCallback MaioRewardedShowCallback
+
+@interface GADRTBMaioInterstitialAd () <MaioInterstitialLoadCallback, MaioInterstitialShowCallback>
+
+@end
+
+@implementation GADRTBMaioInterstitialAd {
+  GADMediationInterstitialLoadCompletionHandler _completionHandler;
+  __weak id<GADMediationInterstitialAdEventDelegate> _adEventDelegate;
+  MaioInterstitial *_interstitial;
+}
+
+- (void)loadInterstitialForAdConfiguration:(nonnull GADMediationInterstitialAdConfiguration *)adConfiguration
+                       completionHandler:(nonnull GADMediationInterstitialLoadCompletionHandler)completionHandler {
+  _completionHandler = completionHandler;
+
+  MaioRequest *request = [[MaioRequest alloc] initWithZoneId:kGADRTBMaioAdapterZoneId testMode:adConfiguration.isTestRequest bidData:adConfiguration.bidResponse];
+  _interstitial = [MaioInterstitial loadAdWithRequest:request callback:self];
+}
+
+- (void)presentFromViewController:(nonnull UIViewController *)viewController {
+  [_interstitial showWithViewContext:viewController callback:self];
+}
+
+#pragma mark - MaioInterstitialLoadCallback, MaioInterstitialShowCallback
+
+- (void)didLoad:(MaioInterstitial *)ad {
+  _adEventDelegate = _completionHandler(self, nil);
+}
+
+- (void)didFail:(MaioInterstitial *)ad errorCode:(NSInteger)errorCode {
+  NSString *description = @"maio open-bidding SDK returned error";
+  NSDictionary *userInfo = @{NSLocalizedDescriptionKey : description, NSLocalizedFailureReasonErrorKey : description};
+  NSError *error = [NSError errorWithDomain:kGADMMaioSDKErrorDomain code:errorCode userInfo:userInfo];
+
+  if (10000 <= errorCode && errorCode < 20000) {
```

Comment:
If I understand correctly, Errors 10000-20000 are load errors, and 20000-30000 are show related errors? If so, can you add some comments here to clarify that?

Additionally, this should be an if/elseif/else in the case the error code that is returned does not belong in these ranges.